.github/workflows/publish-sdist-wheels.yml

```diff
@@ -66,35 +69,41 @@ jobs:
         with:
           submodules: 'true'
 
-      - name: Set up Python 3.6
+      - name: Set up Python 3.7
         uses: actions/setup-python@v2
         with:
-          python-version: 3.6
+          python-version: 3.7
+
       - name: Build a source tarball
         run:
           python setup.py sdist
+
       - name: Test tarball
         run: |
           sudo apt install -y libhdf5-dev
           pip install -r tests/requirement_tests.txt
           pip install dist/*
           pytest -s -v tests
+
       - name: Store sdist as artifact
         uses: actions/upload-artifact@v2
         with:
           name: dist
-          path: dist
+          path: dist/*.whl
 
   upload_wheels:
+    name: Upload wheels to PyPI
+    if: github.event_name == 'release' && github.event.action == 'published'
```

Comment:
Am I correct that this is a replacement for
```yaml
on:
  push:
    tags:
      - 'v[0-9]+.[0-9]+.[0-9]+'
```
from before on the lines 3-6

Reply:
yes, that is the plan - whether it works - I will find out when I try the release.

However, the reasoning is that I want to have the build artifacts to test before they are uploaded to pypi; and they should exist for each commit (FYI).

Reply:
Seems legit. Would not want to have wheel building break and only find out at release time.

Reply:
Yeah, that's is a nightmare that I luckily haven't had to live through - this should be safe now.